src/i_winmusic.c

```diff
@@ -393,10 +394,49 @@ void ResetDevice(void)
 
 boolean I_WIN_InitMusic(void)
 {
-    UINT MidiDevice = MIDI_MAPPER;
+    UINT MidiDevice;
+    int all_devices;
+    int i;
     MIDIHDR *hdr = &buffer.MidiStreamHdr;
+    MIDIOUTCAPS mcaps;
     MMRESULT mmr;
 
+    // find the midi device that matches the saved one
+    if (winmm_midi_device != NULL)
+    {
+        all_devices = midiOutGetNumDevs() + 1; // include MIDI_MAPPER
+        for (i = 0; i < all_devices; ++i)
+        {
+            mmr = midiOutGetDevCaps(i - 1, &mcaps, sizeof(mcaps));
```

Comment:
This will call `midiOutGetDevCaps` with `-1`. Is this right?

Reply:
Yes, MIDI_MAPPER has a device id of (UINT)-1.

Reply:
Please add a comment to that effect so someone read the code can also see that

Reply:
Done.